### PR TITLE
Fix/clock

### PR DESCRIPTION
**Pull request by francesco-romano**

### CODE REVIEW

**Comment by francesco-romano** [comment]
@drdanz branch is protected. I needed to create a PR

**Comment by traversaro** [comment]
Codacy failures are not related to changes in this PR.

File: src/libYARP_OS/src/Time.cpp [hunk 1]
```diff
@@ -81,8 +81,10 @@ static Clock *getClock()
 
 void yarp::os::impl::removeClock()
 {
-    if(pclock != nullptr)
+    if(pclock) {
         delete pclock;
+        pclock = nullptr;
```
Comment (barbalberto) [inline src/libYARP_OS/src/Time.cpp]: Thanks, good catch. Shall we use YARP_NULLPTR?

File: src/libYARP_OS/src/Time.cpp [hunk 2]
```diff
@@ -213,11 +215,15 @@ void Time::useNetworkClock(const ConstString& clock, ConstString localPortName)
 }
 
 void Time::useCustomClock(Clock *clock) {
-    if(clock == nullptr)
+    if(clock == nullptr) {
         YARP_FAIL(Logger::get(), "failed configuring CustomClock client");
+        return;
+    }
```
Comment (barbalberto) [inline src/libYARP_OS/src/Time.cpp]: YARP_FAIL calls `exit`, so `return` here is redundant but made the code more readable (and may static analyzer are more happy)
Shall we update all `YARP_FAIL`s? look line 181 and 196

File: src/libYARP_OS/src/Time.cpp (hunk 1 above)
Comment (traversaro) [reply src/libYARP_OS/src/Time.cpp]: I think now that YARP requires C++11 we can just use `nullptr` everywhere.

Comment (francesco-romano) [reply src/libYARP_OS/src/Time.cpp]: Yes.. furthermore the original code used `nullptr` so I saw no reason to use `YARP_NULLPTR`

File: src/libYARP_OS/src/Time.cpp (hunk 2 above)
Comment (francesco-romano) [reply src/libYARP_OS/src/Time.cpp]: I imagines, but as you say:
- it is not clear for people reading the code
- the static analyzer was not happy! So I preferred being explicit (given that this did not change the behaviour)

Comment (francesco-romano) [reply src/libYARP_OS/src/Time.cpp]: anyway, I added the other two returns

**Comment by traversaro** [comment]
Merging as there is a Travis hiccup preventing the PR to be fully green.

**Pull request merged**